File: Remove Linked List Elements.js
var removeElements = function(head, val) {
    while(head != null && head.val == val)
        head = head.next;
    
    if(head == null)
        return null;
    var post = head.next;
    var cur = head;
    
    while(post)
    {
        if(post.val != val)
        {
            cur.next = post;
            cur = post;
        }
        else
        {
            cur.next = null;
        }
        post = post.next;
    }
    return head;
};

File: Remove Linked List Elements.c
struct ListNode* removeElements(struct ListNode* head, int val) {
        while(head!=NULL && head->val == val)
            head = head->next;
        if(head == NULL)
            return NULL;
        struct ListNode * p = head->next;
        struct ListNode * q = head;
        while(p)
        {
            if(p->val != val)
            {
                q->next = p;
                q = p;
            }
            else
            {
                q->next = NULL;
            }
            p = p->next;
        }
        return head;
}
